test/jsdom/encoding.js

```diff
@@ -0,0 +1,464 @@
+"use strict";
+
+const assert = require("chai").assert;
+const describe = require("mocha-sugar-free").describe;
+const specify = require("mocha-sugar-free").specify;
+const before = require("mocha-sugar-free").before;
+const after = require("mocha-sugar-free").after;
+
+const fs = require("fs");
+const path = require("path");
+const http = require("http");
+const portfinder = require("portfinder");
+const jsdom = require("../..");
+
+describe("jsdom/encoding", { skipIfBrowser: true }, () => {
+  let server;
+  let testHost;
+
+  before(() => {
+    return new Promise((done, rej) => {
+      portfinder.getPort((err, port) => {
+        if (err) {
+          return rej(err);
+        }
+
+        server = http.createServer((req, res) => {
```

Comment:
This is a big ask of you, but could you possibly turn these into to-upstream web-platform-tests, instead of Node specific tests?

For the tests which don't have any custom headers, it shouldn't be too hard: just manually create the right html files.

For those with custom headers, I think it would not be easy because we don't use the Python runner to run those yet, so we can leave those in Node-specific form for now.